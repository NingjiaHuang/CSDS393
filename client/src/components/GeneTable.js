import React, {useState} from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';

import { Button } from 'semantic-ui-react'

const useStyles = makeStyles({
  table: {
    minWidth: 650,
  },
});


export default function GeneTable(){
  const [bs_value, bs_setValue] = useState('N/N');
  const [bd_value, bd_setValue] = useState('N/N');
  const [hs_value, hs_setValue] = useState('N/N');
  const [hd_value, hd_setValue] = useState('N/N');
  const [ps_value, ps_setValue] = useState('N/N');
  const [pd_value, pd_setValue] = useState('N/N');

  const [click_value, click_setValue] = useState("")

  const clickme = async (e) => {
    e.preventDefault()
    try {
        click_setValue([bs_value,bd_value,ps_value,pd_value,hs_value,hd_value]);
        const body = {click_value}
        const response = await fetch("http://localhost:4020/genecalculator", {
            method: "POST",
            headers: {"Content-Type": "application/json"},
            body: JSON.stringify(body)
        })
        const parseRes = await response.json();
    } catch(err) {
        console.error(err.message);
    }
}
  const classes = useStyles();
  



  return (
    
    <TableContainer component={Paper}>
      <Table className={classes.table} aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell>Types</TableCell>
            <TableCell align="center">Sire Genetic Results</TableCell>
            <TableCell align="center">Dam Genetic Results</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
           
            <TableRow>
              <TableCell component="th" scope="row">
                Blood Type
              </TableCell>
              <TableCell align="center">
              <select
                className="custom-select"
                value={bs_value}
                onChange={(e) => {
                  const selectedFood = e.target.value;
                  bs_setValue(selectedFood);
                }}
              >
                <option value="N/N">N/N</option>
                <option value="N/c">N/c</option>
                <option value="N/b">N/b</option>
                <option value="b/b">b/b</option>
                <option value="c/b">c/b</option>
                <option value="c/c">c/c</option>
              </select>
                <p>
                  You selected {bs_value}
                </p>

              </TableCell>
              <TableCell align="center">
                <select
                className="custom-select"
                value={bd_value}
                onChange={(e) => {
                  const selectedFood = e.target.value;
                  bd_setValue(selectedFood);
                }}
              >
                <option value="N/N">N/N</option>
                <option value="N/c">N/c</option>
                <option value="N/b">N/b</option>
                <option value="b/b">b/b</option>
                <option value="c/b">c/b</option>
                <option value="c/c">c/c</option>
              </select>
                <p>
                  You selected {bd_value}
                </p>
              </TableCell>
            </TableRow>


            <TableRow>
              <TableCell component="th" scope="row">
                PDK1
              </TableCell>
              <TableCell align="center">
              <select
                className="custom-select"
                value={ps_value}
                onChange={(e) => {
                  const selectedFood = e.target.value;
                  ps_setValue(selectedFood);
                }}
              >
                <option value="N/N">N/N</option>
                <option value="N/P">N/P</option>
                <option value="P/P">P/P</option>
              </select>
                <p>
                  You selected {ps_value}
                </p>

              </TableCell>
              <TableCell align="center">
              <select
                className="custom-select"
                value={pd_value}
                onChange={(e) => {
                  const selectedFood = e.target.value;
                  pd_setValue(selectedFood);
                }}
              >
                <option value="N/N">N/N</option>
                <option value="N/P">N/P</option>
                <option value="P/P">P/P</option>
              </select>
                <p>
                  You selected {pd_value}
                </p>

              </TableCell>
            </TableRow>

            
            <TableRow>
              <TableCell component="th" scope="row">
                HCM
              </TableCell>
              <TableCell align="center">
              <select
                className="custom-select"
                value={hs_value}
                onChange={(e) => {
                  const selectedFood = e.target.value;
                  hs_setValue(selectedFood);
                }}
              >
                <option value="N/N">N/N</option>
                <option value="N/HCMrd">N/HCMrd</option>
                <option value="HCMrd/HCMrd">HCMrdP/HCMrd</option>
              </select>
                <p>
                  You selected {hs_value}
                </p>

              
              </TableCell>
              <TableCell align="center">
              <select
                className="custom-select"
                value={hd_value}
                onChange={(e) => {
                  const selectedFood = e.target.value;
                  hd_setValue(selectedFood);
                }}
              >
                <option value="N/N">N/N</option>
                <option value="N/HCMrd">N/HCMrd</option>
                <option value="HCMrd/HCMrd">HCMrdP/HCMrd</option>
              </select>
                <p>
                  You selected {hd_value}
                </p>

              </TableCell>
            </TableRow>

            <TableRow>
            <TableCell component="th" scope="row">
                <p>You clicked {click_value}</p>
                <Button 
                data-testid="calBtn"
                content='Calculate' 
                icon='signup' 
                size='sm'  
                style={{ marginLeft: "auto" }} 
                onClick={clickme}>
                </Button>
                <p>
                  You selected Blood Type: {bs_value}, {bd_value}; PKD: {ps_value} ,{pd_value}; HCM: {hs_value} ,{hd_value}
                </p>
            
              </TableCell>
            </TableRow>
  
        </TableBody>
      </Table>
    </TableContainer>

  )
}
