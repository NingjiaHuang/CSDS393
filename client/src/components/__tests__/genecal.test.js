import {render, screen, fireEvent, cleanup} from '@testing-library/react';
import GeneTable from '../GeneTable';
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect';
afterEach(() =>{
    cleanup();
})


test('calcultae button ', () => {
    const mockOnClick = jest.fn()
    const { getByTestId } = render(<GeneTable onClick={mockOnClick()} />)
    const clickIndicator = getByTestId('calBtn')
    fireEvent.click(clickIndicator)
    expect(mockOnClick).toHaveBeenCalledTimes(1)

})


test("Test with mock bloodtype sire", () => {
    render(<GeneTable />);
    expect(screen.getByText("You selected bs N/N")).toBeInTheDocument();
    fireEvent.change(screen.getByTestId("bs"), {
      target: { value: "N/b" },
    });
    expect(screen.getByText("You selected bs N/b")).toBeInTheDocument();
  });

  test("Test with mock bloodtype dam", () => {
    render(<GeneTable />);
    expect(screen.getByText("You selected bd N/N")).toBeInTheDocument();
    fireEvent.change(screen.getByTestId("bd"), {
      target: { value: "N/b" },
    });
    expect(screen.getByText("You selected bd N/b")).toBeInTheDocument();
  });

  test("Test with mock pdk sire", () => {
    render(<GeneTable />);
    expect(screen.getByText("You selected ps N/N")).toBeInTheDocument();
    fireEvent.change(screen.getByTestId("ps"), {
      target: { value: "N/P" },
    });
    expect(screen.getByText("You selected ps N/P")).toBeInTheDocument();
  });

  test("Test with mock pdk dam", () => {
    render(<GeneTable />);
    expect(screen.getByText("You selected pd N/N")).toBeInTheDocument();
    fireEvent.change(screen.getByTestId("pd"), {
      target: { value: "N/P" },
    });
    expect(screen.getByText("You selected pd N/P")).toBeInTheDocument();
  });

  test("Test with mock hcm sire", () => {
    render(<GeneTable />);
    expect(screen.getByText("You selected hs N/N")).toBeInTheDocument();
    fireEvent.change(screen.getByTestId("hs"), {
      target: { value: "N/HCMrd" },
    });
    expect(screen.getByText("You selected hs N/HCMrd")).toBeInTheDocument();
  });

  test("Test with mock hcm dam", () => {
    render(<GeneTable />);
    expect(screen.getByText("You selected hd N/N")).toBeInTheDocument();
    fireEvent.change(screen.getByTestId("hd"), {
      target: { value: "N/HCMrd" },
    });
    expect(screen.getByText("You selected hd N/HCMrd")).toBeInTheDocument();
  });