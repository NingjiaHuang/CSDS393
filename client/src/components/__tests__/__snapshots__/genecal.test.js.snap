// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot of add an gene calculator 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <table
    aria-label="simple table"
    className="MuiTable-root makeStyles-table-1"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Types
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
          scope="col"
        >
          Sire Genetic Results
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
          scope="col"
        >
          Dam Genetic Results
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          role="cell"
          scope="row"
        >
          Blood Type
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
        >
          <select
            className="custom-select"
            onChange={[Function]}
            value="N/N"
          >
            <option
              value="N/N"
            >
              N/N
            </option>
            <option
              value="N/c"
            >
              N/c
            </option>
            <option
              value="N/b"
            >
              N/b
            </option>
            <option
              value="b/b"
            >
              b/b
            </option>
            <option
              value="c/b"
            >
              c/b
            </option>
            <option
              value="c/c"
            >
              c/c
            </option>
          </select>
          <p>
            You selected 
            N/N
          </p>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
        >
          <select
            className="custom-select"
            onChange={[Function]}
            value="N/N"
          >
            <option
              value="N/N"
            >
              N/N
            </option>
            <option
              value="N/c"
            >
              N/c
            </option>
            <option
              value="N/b"
            >
              N/b
            </option>
            <option
              value="b/b"
            >
              b/b
            </option>
            <option
              value="c/b"
            >
              c/b
            </option>
            <option
              value="c/c"
            >
              c/c
            </option>
          </select>
          <p>
            You selected 
            N/N
          </p>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          role="cell"
          scope="row"
        >
          PDK1
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
        >
          <select
            className="custom-select"
            onChange={[Function]}
            value="N/N"
          >
            <option
              value="N/N"
            >
              N/N
            </option>
            <option
              value="N/P"
            >
              N/P
            </option>
            <option
              value="P/P"
            >
              P/P
            </option>
          </select>
          <p>
            You selected 
            N/N
          </p>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
        >
          <select
            className="custom-select"
            onChange={[Function]}
            value="N/N"
          >
            <option
              value="N/N"
            >
              N/N
            </option>
            <option
              value="N/P"
            >
              N/P
            </option>
            <option
              value="P/P"
            >
              P/P
            </option>
          </select>
          <p>
            You selected 
            N/N
          </p>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          role="cell"
          scope="row"
        >
          HCM
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
        >
          <select
            className="custom-select"
            onChange={[Function]}
            value="N/N"
          >
            <option
              value="N/N"
            >
              N/N
            </option>
            <option
              value="N/HCMrd"
            >
              N/HCMrd
            </option>
            <option
              value="HCMrd/HCMrd"
            >
              HCMrdP/HCMrd
            </option>
          </select>
          <p>
            You selected 
            N/N
          </p>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
        >
          <select
            className="custom-select"
            onChange={[Function]}
            value="N/N"
          >
            <option
              value="N/N"
            >
              N/N
            </option>
            <option
              value="N/HCMrd"
            >
              N/HCMrd
            </option>
            <option
              value="HCMrd/HCMrd"
            >
              HCMrdP/HCMrd
            </option>
          </select>
          <p>
            You selected 
            N/N
          </p>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          role="cell"
          scope="row"
        >
          <button
            className="ui sm button"
            data-testid="calBtn"
            onClick={[Function]}
            style={
              Object {
                "marginLeft": "auto",
              }
            }
          >
            <i
              aria-hidden="true"
              className="signup icon"
              onClick={[Function]}
            />
            Calculate
          </button>
        </th>
      </tr>
    </tbody>
  </table>
  <h1>
    Result: 
  </h1>
  <h3>
    
  </h3>
</div>
`;

exports[`matches snapshot of the gene calculator 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <table
    aria-label="simple table"
    className="MuiTable-root makeStyles-table-1"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Types
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
          scope="col"
        >
          Sire Genetic Results
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
          scope="col"
        >
          Dam Genetic Results
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          role="cell"
          scope="row"
        >
          Blood Type
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
        >
          <select
            className="custom-select"
            onChange={[Function]}
            value="N/N"
          >
            <option
              value="N/N"
            >
              N/N
            </option>
            <option
              value="N/c"
            >
              N/c
            </option>
            <option
              value="N/b"
            >
              N/b
            </option>
            <option
              value="b/b"
            >
              b/b
            </option>
            <option
              value="c/b"
            >
              c/b
            </option>
            <option
              value="c/c"
            >
              c/c
            </option>
          </select>
          <p>
            You selected 
            N/N
          </p>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
        >
          <select
            className="custom-select"
            onChange={[Function]}
            value="N/N"
          >
            <option
              value="N/N"
            >
              N/N
            </option>
            <option
              value="N/c"
            >
              N/c
            </option>
            <option
              value="N/b"
            >
              N/b
            </option>
            <option
              value="b/b"
            >
              b/b
            </option>
            <option
              value="c/b"
            >
              c/b
            </option>
            <option
              value="c/c"
            >
              c/c
            </option>
          </select>
          <p>
            You selected 
            N/N
          </p>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          role="cell"
          scope="row"
        >
          PDK1
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
        >
          <select
            className="custom-select"
            onChange={[Function]}
            value="N/N"
          >
            <option
              value="N/N"
            >
              N/N
            </option>
            <option
              value="N/P"
            >
              N/P
            </option>
            <option
              value="P/P"
            >
              P/P
            </option>
          </select>
          <p>
            You selected 
            N/N
          </p>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
        >
          <select
            className="custom-select"
            onChange={[Function]}
            value="N/N"
          >
            <option
              value="N/N"
            >
              N/N
            </option>
            <option
              value="N/P"
            >
              N/P
            </option>
            <option
              value="P/P"
            >
              P/P
            </option>
          </select>
          <p>
            You selected 
            N/N
          </p>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          role="cell"
          scope="row"
        >
          HCM
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
        >
          <select
            className="custom-select"
            onChange={[Function]}
            value="N/N"
          >
            <option
              value="N/N"
            >
              N/N
            </option>
            <option
              value="N/HCMrd"
            >
              N/HCMrd
            </option>
            <option
              value="HCMrd/HCMrd"
            >
              HCMrdP/HCMrd
            </option>
          </select>
          <p>
            You selected 
            N/N
          </p>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
        >
          <select
            className="custom-select"
            onChange={[Function]}
            value="N/N"
          >
            <option
              value="N/N"
            >
              N/N
            </option>
            <option
              value="N/HCMrd"
            >
              N/HCMrd
            </option>
            <option
              value="HCMrd/HCMrd"
            >
              HCMrdP/HCMrd
            </option>
          </select>
          <p>
            You selected 
            N/N
          </p>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          role="cell"
          scope="row"
        >
          <button
            className="ui sm button"
            data-testid="calBtn"
            onClick={[Function]}
            style={
              Object {
                "marginLeft": "auto",
              }
            }
          >
            <i
              aria-hidden="true"
              className="signup icon"
              onClick={[Function]}
            />
            Calculate
          </button>
        </th>
      </tr>
    </tbody>
  </table>
  <h1>
    Result: 
  </h1>
  <h3>
    
  </h3>
</div>
`;
